plugins/backfill/main.py

```diff
@@ -73,37 +77,21 @@ def stream(self):
         dag_name = request.args.get("dag_name")
         start_date = request.args.get("start_date")
         end_date = request.args.get("end_date")
-        clear = request.args.get("clear")
-        dry_run = request.args.get("dry_run")
-        task_regex = request.args.get("task_regex")
-        use_task_regex = request.args.get("use_task_regex")
+        clear = request.args.get("clear").lower() == "true"
+        dry_run = request.args.get("dry_run").lower() == "true"
+        use_task_regex = request.args.get("use_task_regex").lower() == "true"
+        task_regex = request.args.get("task_regex") if use_task_regex else None
 
         # Construct the airflow command
-        cmd = ['airflow']
-
-        if clear == 'true':
-            cmd.append('tasks')
-            cmd.append('clear')
-            if dry_run == 'true':
-                # For dryruns we simply timeout to avoid zombie procs waiting on user input. The output is what we're interested in
-                timeout_list = ['timeout', '60']
-                cmd = timeout_list + cmd
-            elif dry_run == 'false':
-                cmd.append('-y')
-
-            if use_task_regex == 'true':
-                cmd.extend(['-t', str(task_regex)])
-        elif clear == 'false':
-            cmd.append('dags')
-            cmd.append('backfill')
-            cmd.append('--donot-pickle')
-            if dry_run == 'true':
-                cmd.append('--dry-run')
-
-            if use_task_regex == 'true':
-                cmd.extend(['-t', str(task_regex)])
+        backfill_params = BackfillParams(
+            clear=clear,
+            dry_run=dry_run,
+            task_regex=task_regex,
+            dag_name=dag_name,
+            start_date=start_date,
+            end_date=end_date)
 
-        cmd.extend(['-s', str(start_date), '-e', str(end_date), str(dag_name)])
+        cmd = backfill_params.generate_backfill_command()
 
```

Comment:
@haroldwoo do you mind giving this section a second look? Specifically the translation between the string request args and their python type.

Reply:
what is the reason for adding the .lower == true

Reply:
these variables are `str` instead of `bool`. `.lower()` simplifies the number of strings to compare (`"True"`, `"TRUE", `"true"``) and `== "true"` will convert to boolean.